<a href="https://colab.research.google.com/github/vitorjensen/Python_para_Dados/blob/main/Projetos/An%C3%A1lise%20Explorat%C3%B3ria%20de%20Danos%20Informados/An%C3%A1lise_Explorat%C3%B3ria_de_Danos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##### Importando bibliotecas.

In [19]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as s

##### **Objetivo:** Fazer uma análise exploratória de Danos em todos os estados e todo o período de 2024.

##### Os dados de Danos são todas as ocorrências de desastres e eventos de calamidade acontecidas nas regiões do Brasil e que envolveram mortes ou feridos.



##### Importando a base de dados para análise.

In [11]:
df = pd.read_csv("/content/drive/MyDrive/Danos_Informados.csv");

#####Visualizando o dataset

In [ ]:
df.head()

##### **Primeira parte da Análise Exploratória:**
Tratemento:
  - Remover colunas não utilizadas
  - Trocar os valores "0" por "Null"
  - Total de Nulos e Duplicados
  - Total de colunas e linhas

In [ ]:
# Removendo colunas indesejadas
df.columns

index_coluna_dh_outros_afetados = df.columns.get_loc("DH_Outros Afetados") #Pegando a localização da coluna "DH_Outros Afetados"
colunas_para_deletar = df.columns[index_coluna_dh_outros_afetados + 1:] #Definindo o espaço de colunas a serem deletadas após.
df = df.drop(columns=colunas_para_deletar) # Comando Drop para deletar.
display(df.head())

In [ ]:
# Trocando os valores de 0 por Null
df.replace(0, np.nan, inplace=True)
display(df)

In [26]:
# Somando os valores nulos.
df.isnull().sum().sum()

np.int64(36459)

In [28]:
# Identificando valores duplicados
df.duplicated().sum()

np.int64(0)

In [ ]:
df.info()

##### **Resumo do tratamento:**
  - Total de linhas da tabela: 6032
  - Total de colunas após a limpeza: 53 -> 14
  - Total de valores nulos na tabela: 36459
  - Total de valores duplicados: 0

##### **Iniciando segunda parte da Análise Exploratória:**
Verificar:
  - Colunas Categóricas
  - Colunas Númericas
  - A Importância das colunas "DH"